### PR TITLE
Add TimeSeriesSlidingWindow

### PR DESCRIPTION
**Pull request by NielXu**
### Description
This PR attempts to implement the Sliding Window algorithm for time series proposed in issue 22523.

### Implementation
Added a class `TimeSeriesSlidingWindow` that inherits the `BaseCrossValidator`. It implements the Sliding Window algorithm and makes sure the size of the training sets are always the same. The size of the test set is dynamic and can be limited by `max_test_size`.

### Examples
```python
import numpy as np
from sklearn.model_selection._split import TimeSeriesSplit, TimeSeriesSlidingWindow

X = np.array([[1, 2], [3, 4], [1, 2], [3, 4], [1, 2], [3, 4]])
y = np.array([1, 2, 3, 4, 5, 6])
tscv = TimeSeriesSlidingWindow(window_size=2, max_test_size=3)
print(tscv)
for train_index, test_index in tscv.split(X):
    print("TRAIN:", train_index, "TEST:", test_index)
    X_train, X_test = X[train_index], X[test_index]
    y_train, y_test = y[train_index], y[test_index]

# Output
# TimeSeriesSlidingWindow(max_test_size=3, window_size=2)
# TRAIN: [0 1] TEST: [2 3 4]
# TRAIN: [1 2] TEST: [3 4 5]
# TRAIN: [2 3] TEST: [4 5]
# TRAIN: [3 4] TEST: [5]
```

### CODE REVIEW

**Comment by NielXu** [comment]
@AnsonTran Mentioned that we should make test_size fixed too according to the Sliding Window algorithm. I think it makes sense. We need to notice the following:
- We might want to change the name from `max_test_size` to `test_size`
- We need to update the documentation for it
- We need to consider cases such as `[1,2,3,4], window_size=3, test_size=2`, window_size + test_size > sample_size should raise an error
- We need to change `get_n_splits` method to return the correct size
